metadata-service/auth-impl/src/main/java/com/datahub/authorization/AuthorizerChain.java

```diff
@@ -0,0 +1,68 @@
+package com.datahub.authorization;
+
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.common.errors.AuthorizationException;
+
+
+/**
+ * A configurable chain of {@link Authorizer}s executed in series to attempt to authenticate an inbound request.
+ *
+ * Individual {@link Authorizer}s are registered with the chain using {@link #register(Authorizer)}.
+ * The chain can be executed by invoking {@link #authorize(AuthorizationRequest)}.
+ */
+@Slf4j
+public class AuthorizerChain implements Authorizer {
+
+  private final List<Authorizer> authorizers;
+
+  public AuthorizerChain(final List<Authorizer> authorizers) {
+    this.authorizers = Objects.requireNonNull(authorizers);
+  }
+
+  /**
+   * Registers a new {@link Authorizer} at the end of the authorizer chain.
+   *
+   * @param authorizer the authorizer to register
+   */
+  public void register(@Nonnull final Authorizer authorizer) {
+    Objects.requireNonNull(authorizer);
+    authorizers.add(authorizer);
+  }
+
+  @Override
+  public void init(@Nonnull Map<String, Object> authorizerConfig) {
+    // pass.
+  }
+
+  /**
+   * Executes a set of {@link Authorizer}s and returns the first successful authentication result.
+   *
+   * Returns an instance of {@link AuthorizationResult}.
+   */
+  @Nullable
+  public AuthorizationResult authorize(@Nonnull final AuthorizationRequest request) throws AuthorizationException {
+    Objects.requireNonNull(request);
+    for (final Authorizer authorizer : this.authorizers) {
+      try {
+        log.debug(String.format("Executing Authorizer with class name %s", authorizer.getClass().getCanonicalName()));
```

Comment:
nit: don't use String.format inside of log functions. 
log.debug("Executing Authorizer with class name {}", authorizer.getClass().getCanonicalName());